### DISCUSSION TITLE
SQL Relationship FK can not be changed and dose not properly write when editing.

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2396] [Help]
I am having a problem where I can save/edit a row in the DB view. but not the FK only rows that are native to that table.
![image](https://user-images.githubusercontent.com/80058238/129783231-8555902a-715d-40ae-96b0-69381e58da28.png)

however, when I go and save/edit a row in the app it fails. 
these are the errors I'm getting. 
![image](https://user-images.githubusercontent.com/80058238/129783431-cd5ed631-4b7b-4de4-9d4c-6d2bd79de04c.png)

Update I have been playing with this for a day now. I do not believe BB is handling the FK properly. I can not change the FK manually. I have to change the BB-created FK drop-down selector to get the FK to change. I also can not edit a row unless all of the FK has been reselected. this is telling me that BB is not reading the FK correctly and retaining the info when saving the edit.  
Here is a video 

https://user-images.githubusercontent.com/80058238/130155566-8b9eefda-976d-4e6b-85e7-5cc1c680b0a6.mp4

**Comment by mjashanks** [comment]
Hey Andrew, this is a bug which I have logged here: https://github.com/Budibase/budibase/issues/2435

**Reply by KernOil** [reply]
@mjashanks Thank you very much for looking into this!
